net/ip.go

```diff
@@ -9,13 +9,37 @@ import (
 	bmath "github.com/bio-routing/bio-rd/util/math"
 )
 
+var (
+	v4Loopback = NewPfx(IPv4FromOctets(127, 0, 0, 0), 8).Ptr()
+)
+
 // IP represents an IPv4 or IPv6 address
 type IP struct {
 	higher   uint64
 	lower    uint64
 	isLegacy bool
 }
 
+func (ip IP) IsLoopback() bool {
```

Comment:
While you're at it I'd include IPv6 link-local here too (and rename the method to something meaningful)